### DISCUSSION TITLE
ITERATE instead of FOR?

### DISCUSSION THREAD

**Discussion by crocash-dev** [ciel-lang/CIEL#77] [CIEL Ideas]
So I just got CIEL, and it's great, a ton of libraries I'd have spent a long time finding out about, all in one place and suitable for my needs. However I use ITERATE ever since I discovered it early in my Lisp journey. It is really intuitive, self-documented also for when it's not, it works like LOOP but better (because it fits the standard syntax of Lisp and comes naturally to the mind, but also because many things which are awkward in LOOP are trivial in ITERATE, and also because ITERATE is documented.) I think it would be a good addition to CIEL and a replacement for the half-shipped FOR (if people really want to try FOR, then can download it.) I just think ITERATE is *really, really good* and worth considering for permanent inclusion in this wonderful meta-package.

Edit: Although I see a problem: it would conflict with standard CL:LOOP, maybe?

**Comment by vindarel** [comment]
Hi, thanks for the feedback and glad you have a use for CIEL. 

Because of the symbols conflicts we can't enable iterate by default, but I suppose it could be present in the image. Maybe we could even provide a package that has the conflicts resolved.

TBH the reason iterate isn't included is because I use loop. To help in its discoverability I enriched the Iteration page of the Cookbook. But iterating over hash-tables is still not ideal…

I hear that iterate doesn't come without pitfalls, mainly because of being based on code walking. I read good examples, can't find them any more (it was on Discord).

Why do you say For is half-baked?

---

So, I'm not rushing in adding Iterate but not excluding it too. I prefer to take time to test.
